reddit/reddit.go

```diff
@@ -109,3 +112,39 @@ func MaxFeedForCtx(ctx context.Context) int {
 
 	return GuildMaxFeedsNormal
 }
+
+func CheckSubreddit(name string) bool {
+	var redditData struct {
+		Data struct {
+			Dist     int `json:"dist"`
+			Children []struct {
+			}
+		} `json:"data"`
+	}
+
+	query := "https://api.reddit.com/subreddits/search.api?q=" + name
+	req, err := http.NewRequest("GET", query, nil)
+	if err != nil {
+		return false
+	}
+
+	req.Header.Set("User-Agent", UserAgent())
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
```suggestion
	body, err := io.ReadAll(resp.Body)
```
`ioutil` is deprecated.